sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -920,7 +931,7 @@ private[execution] object AggregateProcessor {
     // Create the projections.
     val initialProjection = newMutableProjection(
       initialValues,
-      Seq(SizeBasedWindowFunction.n))()
+      Seq(partitionSize))()
```

Comment:
OK. Thanks! How about we add comment to explain this. This logic is not very easy to understand. Because in most of cases (when `trackPartitionSize=false`), we do not need to bind when we set initial values (because most of them will be constant values), we just put partitionSize at here.

So, the logical is actually the following, right? 
```
val initialProjection =
  if (trackPartitionSize) {
    newMutableProjection(initialValues, Seq(partitionSize))()
  } else {
    newMutableProjection(initialValues, Seq())()
  }
```